crates/server/src/api/middlewares/auth.rs

```diff
@@ -21,6 +24,86 @@ where
     B: MessageBody + 'static,
 {
     from_fn(move |req: ServiceRequest, next: Next<B>| async move {
+        verification(
+            req,
+            &protected_routes,
+            |req, verification_result| async move {
+                match verification_result {
+                    VerificationResult::VerificationPassed(address) => {
+                        {
+                            let mut extensions = req.extensions_mut();
+                            extensions.insert(address);
+                        }
+                        next.call(req).await
+                    }
+                    VerificationResult::VerificationFailed => Err(ErrorUnauthorized("Unathorized")),
+                    VerificationResult::NotProtectedRoute => next.call(req).await,
+                }
+            },
+        )
+        .await
+    })
+}
+
+// This middlware is intended for routes where the auth is optional
+pub fn dcl_optional_auth_middleware<S, B>(
```

Comment:
Both middlewares seem pretty similar, what if we only have one with the two lists? Otherwise, we would run verification twice for each request received